test/SparkConduit.t.sol

```diff
@@ -104,255 +134,719 @@ contract SparkConduitModifierTests is SparkConduitTestBase {
+        assertEq(conduit.totalShares(address(token)), 48 ether);
+
+        assertEq(conduit.requestedShares(address(token), ILK), 0);
+        assertEq(conduit.totalRequestedShares(address(token)), 0);
     }
 
-    function test_withdraw_single_all_liquidity_available() public {
-        conduit.setAssetEnabled(address(token), true);
-        pool.setLiquidityIndex(200_00 * RBPS);
-        conduit.deposit(ILK, address(token), 100 ether);
-        registry.setBuffer(TEST_ADDRESS);
+    function test_withdraw_singleIlk_maxUint() public {
+        _assertTokenState({
+            bufferBalance: 0,
+            atokenBalance: 100 ether
+        });
+
+        _assertATokenState({
+            scaledBalance:     80 ether,
+            scaledTotalSupply: 80 ether,
+            balance:           100 ether,
+            totalSupply:       100 ether
+        });
 
-        assertEq(token.balanceOf(address(atoken)),         100 ether);
-        assertEq(atoken.balanceOf(address(conduit)),       100 ether);
-        assertEq(token.balanceOf(TEST_ADDRESS),            0);
-        assertEq(conduit.getDeposits(ILK, address(token)), 100 ether);
-        assertEq(conduit.getTotalDeposits(address(token)), 100 ether);
+        assertEq(conduit.shares(address(token), ILK), 80 ether);
+        assertEq(conduit.totalShares(address(token)), 80 ether);
+
+        assertEq(conduit.requestedShares(address(token), ILK), 0);
+        assertEq(conduit.totalRequestedShares(address(token)), 0);
 
         vm.expectEmit();
-        emit Withdraw(ILK, address(token), TEST_ADDRESS, 100 ether);
+        emit Withdraw(ILK, address(token), buffer, 100 ether);
         assertEq(conduit.withdraw(ILK, address(token), type(uint256).max), 100 ether);
 
-        assertEq(token.balanceOf(address(atoken)),         0);
-        assertEq(atoken.balanceOf(address(conduit)),       0);
-        assertEq(token.balanceOf(TEST_ADDRESS),            100 ether);
-        assertEq(conduit.getDeposits(ILK, address(token)), 0);
-        assertEq(conduit.getTotalDeposits(address(token)), 0);
-    }
+        _assertTokenState({
+            bufferBalance: 100 ether,
+            atokenBalance: 0
+        });
 
-    function test_withdraw_multi_partial_liquidity_available() public {
-        conduit.setAssetEnabled(address(token), true);
-        pool.setLiquidityIndex(200_00 * RBPS);
+        _assertATokenState({
+            scaledBalance:     0,
+            scaledTotalSupply: 0,
+            balance:           0,
+            totalSupply:       0
+        });
 
-        conduit.deposit(ILK,  address(token), 100 ether);
+        assertEq(conduit.shares(address(token), ILK), 0);
+        assertEq(conduit.totalShares(address(token)), 0);
+
+        assertEq(conduit.requestedShares(address(token), ILK), 0);
+        assertEq(conduit.totalRequestedShares(address(token)), 0);
+    }
+
+    function test_withdraw_multiIlk_exactWithdraw() public {
```

Comment:
`exactWithdraw` is confusing as there is only a partial withdraw of 50 from the 100 available to `ILK`.

Reply:
Updated